### PR TITLE
feat: Add ref to Collapse and Panel component

### LINKED ISSUE #39628: Add ref to Collapse and Panel component
### What problem does this feature solve?

I'm using `Collapse` component with a lot of `Panel` components within it with scrolling. I am currently trying to use third party libraries to achieve virtual scroll. Every library that I looked for used refs in order to make the calculation of heights etc. So in order to achieve virtual scroll I need to forward the ref into the `Collapse` and the `Panel` components.

### What does the proposed API look like?

Making `Collapse` and `Panel` components as `forwardRef` components so they can get ref from outside.

### PR DESCRIPTION
**Pull request by linhf123**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
fix #39628

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       Add ref to Collapse and Panel componen    |
| 🇨🇳 Chinese |      为 Collapse 和 Panel 添加 ref     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by any1024** [comment]
Original PR: [39670](https://github.com/ant-design/ant-design/pull/39670)

**Comment by any1024** [comment]
@afc163 帮忙 cr 下

**Comment by any1024** [comment]
@afc163 帮忙 cr 下

**Comment by any1024** [comment]
@afc163 帮忙 cr 下

**Comment by any1024** [comment]
@afc163 帮忙 cr 下

File: components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 1]
```diff
@@ -594,7 +594,6 @@ exports[`renders ./components/collapse/demo/custom.tsx extend context correctly
 >
   <div
     class="ant-collapse-item ant-collapse-item-active"
-    style="margin-bottom:24px;background:rgba(0, 0, 0, 0.02);border-radius:8px;border:none"
```
Comment (afc163) [inline components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap:597]: 为啥这个会变？

**Comment by any1024** [comment]
`rc-collapse` 下的 Panel 漏掉了 style，我稍后提个 pr 修复下

File: components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap (hunk 1 above)
Comment (any1024) [reply components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap:597]: `rc-collapse` 下的 Panel 漏掉了 style，我稍后提个 pr 修复下

**Review by afc163 — APPROVED** [review]

**Pull request merged**